src/fixedpoint.ts

```diff
@@ -11,17 +25,45 @@ export class FixedInt<T extends number> {
     this.decimals = decimals;
   }
 
+  /**
+   * Get the value of the fixed point number.
+   */
   get val() {
     return this._value;
   }
+
+  /**
+   * Set the value of the fixed point number.
+   *
+   * Does not change the decimals.
+   */
   set val(value: bigint) {
```

Comment:
@tomquirk - I wonder if the public `set val` could be dangerous and lead to unexpected bugs?

Reply:
True. Maybe we should make `val` (and FixedInt generally) immutable. Then if you want to 'change' the val, you need a new instance? @wraeth-eth

Reply:
Yep! This sounds like a good idea to me. Treating them as pods is probably best